test/e2e/util.go

```diff
@@ -856,179 +841,62 @@ func RunRC(config RCConfig) error {
-			pods := item.value.([]*api.Pod)
-			current = len(pods)
-			Logf("%v Controller %s: Found %d pods out of %d", time.Now(), name, current, replicas)
-			if last < current {
-				same = 0
-			} else if last == current {
-				same++
-			} else if current < last {
-				return fmt.Errorf("Controller %s: Number of submitted pods dropped from %d to %d", name, last, current)
-			}
-
-			if same >= failCount {
-				return fmt.Errorf("Controller %s: No pods submitted for the last %d checks", name, failCount)
-			}
-			last = current
-		}
+	interval := config.PollInterval
+	if interval < 0 {
+		interval = 10 * time.Second
 	}
-	if current != replicas {
-		return fmt.Errorf("Controller %s: Only found %d replicas out of %d", name, current, replicas)
-	}
-	Logf("%v Controller %s in ns %s: Found %d pods out of %d", time.Now(), name, ns, current, replicas)
-
-	// Look for all the replicas to be in a Running state.  Stop looking
-	// if all replicas are found in a Running state or no new
-	// replicas are found Running for a continual number of times
-	By(fmt.Sprintf("%v Waiting for all %d replicas to be running with a max container failures of %d", time.Now(), replicas, maxContainerFailures))
-
-	// There must be some amount of pods that have newly transitioned to
-	// the Running state within 100 seconds, so determine the number of
-	// checks needed to ensure timeout within that time period.
-	// 100 seconds is generous amount of time to see a change in the
-	// system even if it is under load.
-	failCount = int(math.Max(1.0, 100.0/float64(interval)))
-
-	same = 0
-	last = 0
-	current = 0
-	var oldPods []*api.Pod
-	podLists.Reset()
-	foundAllPods := false
-	for same < failCount && current < replicas {
-		// Wait just longer than an interval to allow processing
-		// information in the queue quickly
-		time.Sleep(time.Duration(float32(interval)*1.1) * time.Second)
-
-		// Greedily read all existing entries in the queue until
-		// either all pods are running or the queue is empty.  If
-		// the queue is empty we need to stop looking for entries
-		// and wait for a new entry to process
-		for podLists.Len() > 0 && current < replicas {
-			item := podLists.Pop()
-			current = 0
-			waiting := 0
-			pending := 0
-			unknown := 0
-			inactive := 0
-			failedContainers := 0
-			currentPods := item.value.([]*api.Pod)
-			for _, p := range currentPods {
-				if p.Status.Phase == api.PodRunning {
-					current++
-					for _, v := range FailedContainers(p) {
-						failedContainers = failedContainers + v.restarts
-					}
-				} else if p.Status.Phase == api.PodPending {
-					if p.Spec.NodeName == "" {
-						waiting++
-					} else {
-						pending++
-					}
-				} else if p.Status.Phase == api.PodSucceeded || p.Status.Phase == api.PodFailed {
-					inactive++
-				} else if p.Status.Phase == api.PodUnknown {
-					unknown++
+	start := time.Now()
+	running := 0
+	for running < config.Replicas && time.Since(start) < 10*time.Minute {
+		time.Sleep(interval)
```

Comment:
This isn't guaranteed to scale.  30 pods on 10 nodes might start in 10 minutes, but can you guarantee 100 pods on 100 nodes would?  The old looping mechanism didn't care about how long it took, only that there was progress.  Having a hard time limit makes this function less reliable for different pod counts.  I'd prefer going back to tracking progress vs arbitrary timeout